### PR TITLE
[Manta] manta client

### PR DESCRIPTION
**Pull request by zhenfeizhang**

### CODE REVIEW

File: src/extrinsic/manta_pay.rs [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+    /// get the shards
+    async fn get_ledger_state(&self) -> Self::LedgerState {
+        self.coin_shards(None)
+            .await
+            .expect("failed to get ledger state")
+    }
+
+    /// Initialize the assert
+    /// _assert_id is a place holder to support multiple assets in future.
+    async fn init_asset(
```
Comment (stechu) [inline src/extrinsic/manta_pay.rs:143]: Can we have consistent naming with `pallet-manta-pay`, I think init_asset is a better name than `init` though.

File: src/extrinsic/manta_pay.rs [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+                .expect("no transfer event found")
+                .expect("No Event found or decoded");
+            println!("Send {} token from {} to {}", amount, sender, receiver);
+
+            let block_id = transfer_call_events.block;
+            println!("block id: {:?}", block_id);
+        }
+    }
+
+    async fn mint_token(
```
Comment (stechu) [inline src/extrinsic/manta_pay.rs:219]: Also, can we have consistent naming for token/asset?

**Review by stechu — CHANGES_REQUESTED** [review]
Thanks for the PR, this is tons of work!
In general, remove all the dead-code across the files. Also, I highly suggest to have a consistent naming with `pallet-manta-pay`, maybe it is the right time to write a `pallet-manta-pay` API spec.

File: examples/manta_transfer.rs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+// use substrate_subxt::{
+//     balances::*,
+//     system::AccountStoreExt,
+//     ClientBuilder,
+//     MantaNodeRuntime,
+//     PairSigner,
+// };
+
+// #[async_std::main]
+// async fn main() -> Result<(), Box<dyn std::error::Error>> {
```
[the remark this answers is not in the document]
Comment (zhenfeizhang) [reply examples/manta_transfer.rs:29]: yeah... those example... I am working on making them working... so they will be uncommented..

File: src/extrinsic/manta_pay.rs (hunk 1 above)
Comment (zhenfeizhang) [reply src/extrinsic/manta_pay.rs:143]: okay. still with `init_asset` here. will update `pallet-manta-pay`

File: src/extrinsic/manta_pay.rs (hunk 2 above)
Comment (zhenfeizhang) [reply src/extrinsic/manta_pay.rs:219]: lets use `asset`...

File: examples/manta_transfer.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (zhenfeizhang) [reply examples/manta_transfer.rs]: examples re-enabled

**Review by stechu — APPROVED** [review]

**Pull request merged**